app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageActivity.java

```diff
@@ -106,6 +109,21 @@ public void onPageScrollStateChanged(int state) {
       }
     });
 
+    // Disable scrolling for the AppBarLayout on top of the screen
+    // User can only scroll the PageViewer component
+    AppBarLayout appBarLayout = (AppBarLayout) findViewById(R.id.appbar);
+    if (appBarLayout.getLayoutParams() != null) {
+      CoordinatorLayout.LayoutParams layoutParams = (CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams();
+      AppBarLayout.Behavior appBarLayoutBehaviour = new AppBarLayout.Behavior();
+      appBarLayoutBehaviour.setDragCallback(new AppBarLayout.Behavior.DragCallback() {
```

Comment:
Can you convert this to lambda if possible ?

Reply:
Sure. I will send a new PR.

Reply:
@siddharth2010 Please update in this PR only. :)

Reply:
@srv-twry, I tried but couldn't really figure out how the code will be converted to lambda. Can you please help me with it...

Here's the code:

```java
AppBarLayout appBarLayout = (AppBarLayout) findViewById(R.id.appbar);
    if (appBarLayout.getLayoutParams() != null) {
      CoordinatorLayout.LayoutParams layoutParams = (CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams();
      AppBarLayout.Behavior appBarLayoutBehaviour = new AppBarLayout.Behavior();
      appBarLayoutBehaviour.setDragCallback(new AppBarLayout.Behavior.DragCallback() {
        @Override
        public boolean canDrag(@NonNull AppBarLayout appBarLayout) {
          return false;
        }
      });
      layoutParams.setBehavior(appBarLayoutBehaviour);
    }
```
I think that the new code should be something like this:
```java
AppBarLayout appBarLayout = (AppBarLayout) findViewById(R.id.appbar);
    if (appBarLayout.getLayoutParams() != null) {
      CoordinatorLayout.LayoutParams layoutParams = (CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams();
      AppBarLayout.Behavior appBarLayoutBehaviour = new AppBarLayout.Behavior();
      appBarLayoutBehaviour.setDragCallback(v -> return false);
      layoutParams.setBehavior(appBarLayoutBehaviour);
    }
```
This is of course wrong as the target type of the lambda must be an interface
The line:
```java
appBarLayoutBehaviour.setDragCallback(v -> return false);
```
needs to be replaced with something different. Please help me here
Thanks a lot!

Reply:
@siddharth2010  Changing anonymous classes to lambda expressions is not a big deal.
I can be done in android studio very easily.Just put cursor in anonymous class 
and press " alt + enter " , then studio will give an option to change the class into lambda expression,click on it. Task completed.

Hope this helps you.

Reply:
I will check it out right away. Thanks!

Reply:
@siddharth2010 Any updates ?

Reply:
Sorry but i had an exam at college today. I am on my way back and after i reach, i will finish this.

Reply:
I was having some issues. I was wondering if you could help me a bit here. you see, I might be really wrong here, but I feel that this particular anonymous class cannot be converted to lambda
thats because this is a class that is inheriting from a class that was predefined in the sdk. and I am also overriding a predefined function in the sdk...
even android studio code inspector doesn't show this in the "**convert anonymous classes to lambda analysis**" , even though it is showing it in the "**anonymous inner classes list analysis**"... I have read the documentation but I am still not able to figure this out...
Please help me here.

Reply:
@srv-twry , please look into this issue.

Reply:
I checked locally, I don't think it can be converted to lambda. I am not so confident with lambda expressions. @mhutti1 Can you please check and confirm ?

Reply:
Yeah looks like it can't be.